components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java

```diff
@@ -722,7 +722,18 @@ public void deleteTenantRole(int tenantId, String role, String tenantDomain)
                 tenantDomain);
         for (IdentityProvider identityProvider : identityProviders) {
             String identityProviderName = identityProvider.getIdentityProviderName();
-            identityProvider = this.getIdPByName(null, identityProviderName, tenantId, tenantDomain);
+            try {
+                identityProvider = this.getIdPByName(null, identityProviderName, tenantId, tenantDomain);
+            } catch (IdentityProviderManagementException e) {
```

Comment:
we can narrow this Exception to IdentityProviderManagementClientException

Reply:
Fixed with b34720cee85db026d3b6f5f9242a845afd424831